### DISCUSSION TITLE
Visualizer with typescript

### DISCUSSION THREAD

**Discussion by prbxr** [statelyai/xstate#2409] [Q&A] [answered]
Apparently visualizer does not work with typescript.
It does not work with any syntax other than Machine<>.
It will accept state only as a string, for example not as a StateVariants.stateA.

Will it be improved in next release?

**Comment by christoph-fricke** [comment, marked as answer]
As far as I am aware, there is a huge update for the visualizer in the works. In the meantime, I suggest to use the inspector, which is more modern than the visualizer and supports newer features. Best place to get started is https://statecharts.io and the inspector package https://xstate.js.org/docs/packages/xstate-inspect/.

**Comment by Andarist** [comment]
I believe that a new version of the visualizer (that should be released pretty soon) has TS support so the answer would be - yes :tada: it will be improved in the next release

**Comment by lishine** [comment]
Thanks guys!
I was actually confused and confused you.
I called thought that the only way to visualize is to use statecharts.io ...
Also I called it Visualizer ...
So I discovered the inspect package! And that is very cool!

**Reply by lishine** [reply]
ye that is my pesnoal user
